processor/src/test/java/org/mapstruct/ap/test/multisource/MultiSourceMapperTest.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.multisource;
+
+import java.util.Collections;
+
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.ProcessorTest;
+import org.mapstruct.ap.testutil.WithClasses;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Filip Hrisafov
+ */
+public class MultiSourceMapperTest {
```

Comment:
Is it really reproducing the problem as described on #2005? That's a combination of bean, array right?

Reply:
It goes a step further. I started like in #2005. However, I realised that it was a more general multi source parameter mapping. We also have similar problems when the first parameter is a primitive